src/Microsoft.ApplicationInsights/Extensibility/Implementation/TelemetryConfigurationFactory.cs

```diff
@@ -465,5 +462,31 @@ private static IEnumerable<XElement> GetPropertyDefinitions(XElement instanceDef
 
             return attributeDefinitions.Concat(elementDefinitions);
         }
+
+        private void SelectInstrumentationKey(TelemetryConfiguration configuration)
+        {
+            if (PlatformSingleton.Current.TryGetEnvironmentVariable(ConnectionStringEnvironmentVariable, out string connectionStringEnVar))
+            {
+                // TODO: LOG TO ETW INFORMATION: Connection String Environment Variable detected. 
+                configuration.SetConnectionString(connectionStringEnVar);
+            }
+            else if (PlatformSingleton.Current.TryGetEnvironmentVariable(InstrumentationKeyEnvironmentVariable, out string instrumentationKeyEnVar))
+            {
+                // TODO: LOG TO ETW INFORMATION: Instrumentation Key Environment Variable detected. 
+                configuration.InstrumentationKey = instrumentationKeyEnVar;
+            }
+            else if (configuration.ConnectionString != null)
+            {
+                // A set ConnectionString will have come from parsing the config file.
+                // Call Set() to force the connection string to be parsed.
+                configuration.SetConnectionString(configuration.ConnectionString);
```

Comment:
is it setting the same value that is already set?

Reply:
This is needed for the case where Reflection overwrites the Ikey.

Requirements state that a config Connection String should always overwrite the config Instrumentation Key.

If the customer's config has IKey last, it will overwrite the Connection String and I need this to set it back.

Reply:
But I mean you take the value as it as set it as is. If this is needed, it does not look right

```csharp
            else if (configuration.ConnectionString != null)
                configuration.SetConnectionString(configuration.ConnectionString);
```

Reply:
Ah, I guess if ConnectionString setter is the same as SetConnectionString, it will be handled there

Reply:
I agree this doesn't look right, but it's very much needed even if i move to setters!

The edge case I'm solving is that a customer can specify either property in any order in their config file.
Whichever one is last will overwrite the first.
But our requirements are that the Connection String should always take precedence, regardless of the order specified in the config file.

I've captured this in these test cases: https://github.com/microsoft/ApplicationInsights-dotnet/blob/12cff074bb047b8ebbf416ca7bf400de51784c40/Test/Microsoft.ApplicationInsights.Test/Shared/Extensibility/TelemetryConfigurationFactoryTest.cs#L171-L207

Reply:
> The edge case I'm solving is that a customer can specify either property in any order in their config file. Whichever one is last will overwrite the first.

May be fix this instead and do not allow setting ikey if ConnectionString is set?